### PR TITLE
Do not increase download counters if stats=0 is passed.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/505/

File: internal/v4/archive_test.go [hunk 1]
```diff
@@ -119,6 +119,25 @@ func (s *ArchiveSuite) TestGetCounters(c *gc.C) {
...
+	// Download the charm archive using the API, passing stats=0.
+	rec := storetesting.DoRequest(c, storetesting.DoRequestParams{
+		Handler: s.srv,
+		URL:     storeURL(url.Path() + "/archive?stats=0"),
+	})
+	c.Assert(rec.Code, gc.Equals, http.StatusOK)
+
+	// Check that the downloads count for the entity has not been updated.
+	key := []string{params.StatsArchiveDownload, "utopic", "mysql", "", "42"}
+	stats.CheckCounterSum(c, s.store, key, false, 0)
```
Comment (rogpeppe) [inline internal/v4/archive_test.go:138]: Since counter incrementing is asynchronous, is this actually necessarily testing anything?

**Comment by rogpeppe** [comment]
LGTM except that I'm not sure the test is correct.

File: internal/v4/archive_test.go (hunk 1 above)
Comment (frankban) [inline internal/v4/archive_test.go:138]: The CheckCounterSum has some retry logic in it if the expected value is 0.

**Comment by rogpeppe** [comment]
Great, LGTM still.

File: internal/charmstore/stats.go [hunk 2]
```diff
@@ -158,8 +158,11 @@ func timeToStamp(t time.Time) int32 {
 // IncCounterAsync increases by one the counter associated with the composed
 // key. The action is done in the background using a separate goroutine.
 func (s *Store) IncCounterAsync(key []string) {
-	// TODO frankban 2014-09-15: log possible IncCounter errors.
-	go s.IncCounter(key)
+	go func() {
+		if err := s.IncCounter(key); err != nil {
```
Comment (mhilton) [inline internal/charmstore/stats.go:162]: I assume IncCounter is threadsafe.

**Comment by mhilton** [comment]
:+1:

**Comment by frankban** [comment]
:shipit: 
Thanks for the reviews!

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**